okhttp/src/main/java/okhttp3/internal/tls/OkHostnameVerifier.kt

```diff
@@ -51,8 +52,10 @@ object OkHostnameVerifier : HostnameVerifier {
 
   /** Returns true if [certificate] matches [ipAddress]. */
   private fun verifyIpAddress(ipAddress: String, certificate: X509Certificate): Boolean {
+    val canonicalIpAddress = ipAddress.toCanonicalHost()
```

Comment:
FWIW I didn't consider IP Addresses a host path to optimise too much.  But that's certainly debatable.